eureka-client/src/main/java/com/netflix/appinfo/ApplicationInfoManager.java

```diff
@@ -219,6 +219,21 @@ public void refreshDataCenterInfoIfRequired() {
             builder.setHostName(newAddress).setIPAddr(newIp).setDataCenterInfo(config.getDataCenterInfo());
             instanceInfo.setIsDirty();
         }
+
+        String existingSpotInstanceAction  = instanceInfo.getMetadata().get(AmazonInfo.MetaDataKey.spotInstanceAction.name);
+        DataCenterInfo dataCenterInfo = config.getDataCenterInfo();
+        if (dataCenterInfo instanceof AmazonInfo) {
+            AmazonInfo amazonInfo = (AmazonInfo) dataCenterInfo;
+            String newSpotInstanceAction = amazonInfo.getMetadata().get(AmazonInfo.MetaDataKey.spotInstanceAction.name);
+            if (newSpotInstanceAction != null && !newSpotInstanceAction.equals(existingSpotInstanceAction)) {
+                logger.warn(String.format("The spot instance termination action changed from: %s => %s",
+                        existingSpotInstanceAction,
+                        newSpotInstanceAction));
+                InstanceInfo.Builder builder = new InstanceInfo.Builder(instanceInfo);
+                builder.setDataCenterInfo(config.getDataCenterInfo());
```

Comment:
There is duplication of code here with code on line 218/219/220, mind consolidating them? Thanks.

Reply:
Fixed.